mne/io/curry/curry.py

```diff
@@ -0,0 +1,244 @@
+
+    else:
+        curry_vers = 7
+        for check_ext in [".dap", ".dat", ".rs3"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+    return fname_base, curry_vers
+
+
+def _read_curry_info(fname_base, curry_vers):
+    #####################################
+    # read parameters from the param file
+
+    if curry_vers == 7:
+        file_extension = '.dap'
+    else:
+        file_extension = '.cdt.dpa'
+
+    var_names = ['NumSamples', 'NumChannels', 'NumTrials', 'SampleFreqHz',
+                 'TriggerOffsetUsec', 'DataFormat', 'SampleTimeUsec',
+                 'NUM_SAMPLES', 'NUM_CHANNELS', 'NUM_TRIALS', 'SAMPLE_FREQ_HZ',
+                 'TRIGGER_OFFSET_USEC', 'DATA_FORMAT', 'SAMPLE_TIME_USEC']
+
+    param_dict = dict()
+    with open(fname_base + file_extension) as fid:
+        for line in fid:
+            if any(var_name in line for var_name in var_names):
+                key, val = line.replace(" ", "").replace("\n", "").split("=")
+                param_dict[key] = val
+
+    if not len(param_dict) == 7:
+        raise KeyError("Some variables cannot be found in the parameter file.")
+
+    if "NumSamples" in param_dict:
+        n_samples = int(param_dict["NumSamples"])
+        # n_ch = int(param_dict["NumChannels"])
+        n_trials = int(param_dict["NumTrials"])
+        sfreq = float(param_dict["SampleFreqHz"])
+        offset = float(param_dict["TriggerOffsetUsec"]) / 1e6  # convert to s
+        time_step = float(param_dict["SampleTimeUsec"]) / 1e6
+        # data_format = param_dict["DataFormat"]
+
+    else:
+        n_samples = int(param_dict["NUM_SAMPLES"])
+        # n_ch = int(param_dict["NUM_CHANNELS"])
+        n_trials = int(param_dict["NUM_TRIALS"])
+        sfreq = float(param_dict["SAMPLE_FREQ_HZ"])
+        offset = float(param_dict["TRIGGER_OFFSET_USEC"]) / 1e6
+        time_step = float(param_dict["SAMPLE_TIME_USEC"]) / 1e6
+        # data_format = param_dict["DATA_FORMAT"]
+
+    if (sfreq == 0) and (time_step != 0):
+        sfreq = 1. / time_step
+
+    #####################################
+    # read labels from label files
+
+    if curry_vers == 7:
+        file_extension = '.rs3'
+    else:
+        file_extension = '.cdt.dpa'
+
+    ch_names = []
+    ch_pos = []
+
+    save_labels = False
+    save_ch_pos = False
+    with open(fname_base + file_extension) as fid:
+        for line in fid:
+
+            if re.match("LABELS.*? END_LIST", line):
+                save_labels = False
+
+            # if "SENSORS END_LIST" in line:
+            if re.match("SENSORS.*? END_LIST", line):
+                save_ch_pos = False
```

Comment:
Maybe `save_ch_pos = not re.match(...`

Reply:
that wouldn't really work...
The file looks something like this:
```
SOME STUFF

LABELS START_LIST
C3
C4
Cz
LABELS END_LIST

MORE STUFF
```

So, when `LABELS START_LIST` appears, `save_labels` will be constantly set to True and keep saving the lines, until `LABELS END_LIST` appears, where it will be set back to False.
This wouldn't work if you update `save_labels` for each line.

But I'm not sure if i got you right... maybe there's a more efficient way to implement this?

Reply:
* same goes for `save_ch_pos` ofc

Reply:
This pattern of 
```sh
bla bla bla

FOO START[_LIST]
foo
foo
foo
FOO END[_LIST]
```

Appears in many places. You should make a helper function that gets the **token** and returns whatever is in between (and something that consumes until **token**). Then this should pass:

```py

def get_foo_form_file(fname):
    with open(fname) as ff:
        _consume_until_token(ff, token='FOO')
        data = _get_whatever(ff)
    return whatever, ff

if __name__== "__main__":
  asssert get_foo_from_file('path_to_previous_snippet') == 'foo\nfoo\nfoo\n'
```